cpu/efm32_common/include/periph_cpu.h

```diff
@@ -0,0 +1,339 @@
+/*
+ * Copyright (C) 2015-2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_efm32_common
+ * @{
+ *
+ * @file
+ * @brief       CPU specific definitions for internal peripheral handling
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Bas Stottelaar <basstottelaar@gmail.com>
+ */
+
+#ifndef PERIPH_CPU_H
+#define PERIPH_CPU_H
+
+#include "mutex.h"
+
+#include "cpu_conf.h"
+
+#include "em_adc.h"
+#include "em_cmu.h"
+#include "em_device.h"
+#include "em_gpio.h"
+#include "em_usart.h"
+#ifdef _SILICON_LABS_32B_PLATFORM_1
+#include "em_dac.h"
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Enable support for Low-power peripherals (if supported by CPU).
+ * @{
+ */
+#ifndef LOW_POWER_ENABLED
+#define LOW_POWER_ENABLED   (1)
+#endif
+/** @} */
+
+/**
+ * @brief   Internal macro for combining ADC resolution (x) with number of
+ *          shifts (y).
+ */
+#define ADC_MODE(x, y)     ((y << 4) | x)
+
+/**
+ * @brief   Possible ADC resolution settings
+ * @{
+ */
+#define HAVE_ADC_RES_T
+typedef enum {
+    ADC_RES_6BIT = ADC_MODE(adcRes6Bit, 0),     /**< ADC resolution: 6 bit */
+    ADC_RES_8BIT = ADC_MODE(adcRes8Bit, 0),     /**< ADC resolution: 8 bit */
+    ADC_RES_10BIT = ADC_MODE(adcRes12Bit, 2),   /**< ADC resolution: 10 bit (shifted from 12 bit) */
+    ADC_RES_12BIT = ADC_MODE(adcRes12Bit, 0),   /**< ADC resolution: 12 bit */
```

Comment:
the `ADC_RES_14BIT` and `ADC_RES_16BIT` values should also be defined, as these are basically part of the interface...

Reply:
Added and therefore fixed.